### DISCUSSION TITLE
Outdated smoothness and tracktype after surface updates

### DISCUSSION THREAD

**Discussion by sjakobi** [streetcomplete/StreetComplete#4178] [Feedback]
When users solve surface tasks, other surface-related tags of the same object may become outdated.

For example, in https://www.openstreetmap.org/way/60840215/history, version 5 changed the surface from grass to asphalt. But the tags smoothness=very_horrible and tracktype=grade3 remain.

I wonder whether this app could be a bit smarter about these updates.

For example, if the surface is changed from some kind of unpaved surface to a paved surface type, I think the smoothness can be safely deleted.

The tracktype can be checked for consistency with the surface. For example, grade4 and grade5 are fully incompatible with paved surface types.

**Comment by sjakobi** [comment]
I wrote the text above before finding https://github.com/streetcomplete/StreetComplete/discussions/2771 and https://github.com/streetcomplete/StreetComplete/discussions/4063.

Not sure if I'm providing any new ideas here, so feel free to close.

In any case I'm excited for https://github.com/streetcomplete/StreetComplete/pull/4105! :)

**Comment by matkoniecz** [comment]
Removing smoothness if surface value changed makes sense, I think. Even without asking user to confirm.

**Reply by sjakobi** [reply]
Actually, I think that certain changes of the surface _value_ don't necessarily imply that the actual road surface has changed. For example, when `unpaved` is changed to some specific unpaved surface type.

There's also the issue that `compacted`, `fine_gravel`, `gravel` and `pebblestones` can be very hard to distinguish, so changes between these values may not indicate an actual change on the ground.
